### PR TITLE
Clarify html form

### PR DESCRIPTION
**Pull request by avermeulen**
@SteveBarnett please review and merge if 👍

### CODE REVIEW

**Comment by SteveBarnett** [comment]
For the next PR, could you do it on a branch from this repo, rather than yours? That way I can push tiny fixes myself rather than adding little comments all over the show. Ta! :)

File: steps/handlebars.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+---
+# Templates
+
+To combine data and html we use a template engine.
+
+## Templating using Handlebars example
+
+We will be using `handlebars` as our template engine, it combines data with templates to render information.
+
+A typical handlebars template look likes this
```
Comment (SteveBarnett) [inline steps/handlebars.md:12]: A typical handlebars template **looks like** this

File: steps/handlebars.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+</div>{% endraw %}
+{% endhighlight %}
+
+combining the above template with this data:
+
+```javascript
+var user = {
+ author : true,
+ firstName : "Jo",
+ lastName : "Blogss"
```
Comment (SteveBarnett) [inline steps/handlebars.md:29]: Bloggs

File: steps/handlebars.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+We will be using `handlebars` as our template engine, it combines data with templates to render information.
+
+A typical handlebars template look likes this
+
+{% highlight html %}
+{% raw %}
+<div class="entry">
+  {{#if author}}
+    <h1> {{firstName}} {{lastName}}</h1>
```
Comment (SteveBarnett) [inline steps/handlebars.md:18]: `<h1>{{firstName}} {{lastName}}</h1>`

(stray space)

File: steps/handlebars.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+{% highlight html %}
+{% raw %}
+<div class="entry">
+    <h1>Joe Bloggs</h1>
+</div>{% endraw %}
+{% endhighlight %}
+
+## Templating using Handlebars
+
+[HandlebarsJS](https://www.npmjs.com/package/handlebars) is built on top of [Mustache](https://www.npmjs.com/package/mustache) templating engine and extends it. Mustache templating aims to be logic-less templating, but that make Mustache hard to use at times.
```
Comment (SteveBarnett) [inline steps/handlebars.md:44]: built on top of **the** Mustache ...

that **makes** Mustache

File: steps/handlebars.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+Read more on [the Mustache website](https://mustache.github.io/mustache.5.html).
+
+Read here about [what are the differences between Mustache.js and Handlebars.js?](http://stackoverflow.com/questions/10555820/what-are-the-differences-between-mustache-js-and-handlebars-js)
+
+## Handlebars helpers
+
+Handlebars comes with a set of built in helpers that makes it easy for one to process data and convert it into a layout. It [supports things like if statements and loops](http://handlebarsjs.com/builtin_helpers.html).
```
Comment (SteveBarnett) [inline steps/handlebars.md:52]: that make it easy for you to process data

File: steps/handlebars_with_express.md [hunk 6]
```diff
@@ -0,0 +1,21 @@
+---
+layout: default
+---
+
+# View Engine
+
+To make it easy to use Handlebars templates in ExpressJS we configure Handlebars as a view engine in ExpressJS. Using the [express-handlebars](https://www.npmjs.com/package/express-handlebars) module.
```
Comment (SteveBarnett) [inline steps/handlebars_with_express.md:7]: ExpressJS**, u**sing

File: steps/http_request.md [hunk 7]
```diff
@@ -35,7 +35,18 @@ have query parameters that can be read like this:
 
 ### req.body
 
-To read parameters send from HTML forms one use the `req.body` paramater on the Http Request object sent into routes.
+To read parameters sent from HTML forms to the server you use the `body` attribute on the HttpRequest object sent into routes.
+
+For a html form that looks like this in html:
```
Comment (SteveBarnett) [inline steps/http_request.md:40]: The HTML form might look like this:

**Comment by avermeulen** [comment]
Oops that was not intentional. Most Surely.

File: steps/http_request.md [hunk 8]
```diff
@@ -35,7 +35,18 @@ have query parameters that can be read like this:
...
+
+```html
+<form action="/add_product" method="POST">
+    Product name : <input type="text" name="product_name" >
```
Comment (SteveBarnett) [inline steps/http_request.md:44]: `<label>Product name: <input type="text" name="product_name" ></label>`

Not sure whether to do that or 

`<label for="product_name">Product name:</label> <input type="text" name="product_name" id="product_name">`

I think the first one.

File: steps/templating.md [hunk 9]
```diff
@@ -3,71 +3,12 @@ layout: default
 ---
 # Templates
 
-To combine data and html we use a template engine.
+To make creating views in Express one can use various different template engines; to makes it easy to combine data into html files. We will use Handlebars as our template engine.
```
Comment (SteveBarnett) [inline steps/templating.md:6]: To create views in Express you one can use various different template engines; this makes it easy to combine data into HTML files. We will use Handlebars as our template engine.

**Comment by SteveBarnett** [comment]
@avermeulen Have left a monkeyload of comments. Enjoy! :D

**Pull request merged**